### PR TITLE
Feature/ranking screen

### PR DESCRIPTION
**Pull request by nino-gerb**
Screenshot 2023-05-15 072613

A screen to show to ranking of the participants

### CODE REVIEW

File: app/src/main/java/ch/epfl/sdp/cook4me/ui/challenge/RankingScreen.kt [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+                Row(
+                    modifier = Modifier
+                        .fillMaxWidth()
+                        .padding(16.dp)
+                        .animateContentSize(),
+                    horizontalArrangement = Arrangement.Start,
+                    verticalAlignment = Alignment.CenterVertically
+                ) {
+                    Text(
+                        text = "Place",
```
Comment (pauromeu) [inline app/src/main/java/ch/epfl/sdp/cook4me/ui/challenge/RankingScreen.kt:62]: Important: Use string resource so the code is more mainteinable.

File: app/src/main/java/ch/epfl/sdp/cook4me/ui/challenge/RankingScreen.kt [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.unit.dp
+import ch.epfl.sdp.cook4me.R
+import ch.epfl.sdp.cook4me.ui.challengeform.Challenge
+
+const val INCREMENT = 1
+const val MAX = 3
+const val WIDTH_SCORE = 0.7f
+const val WIDTH_PARTICIPANT = 3f
+const val WIDTH_PLACE = 1f
```
Comment (pauromeu) [inline app/src/main/java/ch/epfl/sdp/cook4me/ui/challenge/RankingScreen.kt:35]: Proposition: It is great that you use these constant values inseted of magic numbers! I would also recommend to make them private since they are only used in this file.

File: app/src/main/java/ch/epfl/sdp/cook4me/ui/challenge/RankingScreen.kt [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+        }
+
+        Button(
+            onClick = onBack,
+            modifier = Modifier
+                .align(Alignment.CenterHorizontally)
+                .fillMaxWidth()
+                .padding(16.dp, 8.dp)
+        ) {
+            Text("Back")
```
Comment (pauromeu) [inline app/src/main/java/ch/epfl/sdp/cook4me/ui/challenge/RankingScreen.kt:96]: Important: Same here, use string resource :)

File: app/src/main/java/ch/epfl/sdp/cook4me/ui/challenge/RankingScreen.kt [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+        Row(
+            modifier = Modifier.weight(WIDTH_PARTICIPANT)
+        ) {
+            Text(
+                text = "$participant ",
+                style = MaterialTheme.typography.h6,
+                modifier = Modifier.align(Alignment.CenterVertically)
+            )
+
+            if (isWinner) {
```
Comment (pauromeu) [inline app/src/main/java/ch/epfl/sdp/cook4me/ui/challenge/RankingScreen.kt:128]: Observation: Nice detail! It is great to distinguish the winner with an icon :)

File: app/src/androidTest/java/ch/epfl/sdp/cook4me/ui/challenge/RankingScreenTest.kt [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+import androidx.compose.ui.test.performClick
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import ch.epfl.sdp.cook4me.ui.challengeform.Challenge
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+import java.util.Calendar
+
+@RunWith(AndroidJUnit4::class)
+class RankingScreenTest {
```
Comment (pauromeu) [inline app/src/androidTest/java/ch/epfl/sdp/cook4me/ui/challenge/RankingScreenTest.kt:15]: Question: What happens when there are two winners? Maybe it would be interesting to test this edge case.

**Review by pauromeu — APPROVED** [review]
Hey Nino! Thank you for your contribution to the app. I have seen that some of the changes are the same of the previous PR, that may be because they are still not merged in the main. I have not added any comment on these repeated files, you can find them in the corresponding PR. Overall, it looks great! I love the ranking screen and you have taken into account some details like putting a trophy icon next to the winner, I love it. I just added some comments to make the code a bit more mainteinable :)
